packages/core/src/tracing/span.ts

```diff
@@ -115,30 +116,27 @@ export class Span implements SpanInterface {
    */
   public instrumenter: Instrumenter;
 
+  /**
+   * The origin of the span, giving context about what created the span.
+   */
+  public origin?: SpanOrigin;
+
   /**
    * You should never call the constructor manually, always use `Sentry.startTransaction()`
    * or call `startChild()` on an existing span.
    * @internal
    * @hideconstructor
    * @hidden
    */
-  public constructor(spanContext?: SpanContext) {
-    this.traceId = uuid4();
-    this.spanId = uuid4().substring(16);
-    this.startTimestamp = timestampInSeconds();
-    this.tags = {};
-    this.data = {};
-    this.instrumenter = 'sentry';
-
-    if (!spanContext) {
-      return this;
-    }
-    if (spanContext.traceId) {
-      this.traceId = spanContext.traceId;
-    }
-    if (spanContext.spanId) {
-      this.spanId = spanContext.spanId;
-    }
+  public constructor(spanContext: SpanContext = {}) {
+    this.traceId = spanContext.traceId || uuid4();
+    this.spanId = spanContext.spanId || uuid4().substring(16);
+    this.startTimestamp = spanContext.startTimestamp || timestampInSeconds();
+    this.tags = spanContext.tags || {};
+    this.data = spanContext.data || {};
+    this.instrumenter = spanContext.instrumenter || 'sentry';
+    this.origin = spanContext.origin || 'manual';
+
```

Comment:
l: (or no action required) we might wanna streamline some of the other assignments here as well but it's ouf of scope for this PR. feel free to ignore